submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,225 @@
+  to the topology that we use now. But it will have a single `global` zone assigned to it with a transit switch which is not involved in any
+  packet forwarding. Though this setup is expected to work by default, the subnet that the transit switch used overlaps with the subnet
+  Submariner uses for the logical router. In this proposal, we plan to remove the submariner router and the associated switches and this will
+  not be a problem anymore. The below indicates node annotations in a single zone  OVN setup.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster2-worker
+```
+
+* Multiple Zone: In a multiple-zone setup, we will have an OVN database and a master pod for each zone. Transit switches connect the
+  zones. The OVN-Kubernetes services ensure that the necessary routes are added for pod and service reachability across nodes in different
+  zones.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.3/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.5/16"]'
+      k8s.ovn.org/ovn-zone: az2
+    name: cluster2-worker
+```
+
+With the current architecture, Submariner adds routes only in the zone in which it is deployed. For example, if Submariner is deployed in
+zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk to other clusters. Pods in zone 2 or zone 3 will not
+be able to reach remote clusters connected via Submariner.
+
+### Route APIs
+
+As part of this proposal, we are planning to add two new CRDs in Submariner
+
+#### SubmarinerGwRoute
+
+This CR will be created when a remote endpoint is added and there will be one CR per remote cluster. This crd has two fields
+
+* NextHop - Specifies the next hop to reach the remote cluster.
+* RemoteCIDR - Specifies the list of  remote CIDRs reachable via this cluster, in this case it will be the IP of ovn-k8s-mp0
+interface.
+
+This CR will be used by the gateway pod to program OVN to send the traffic destined to remote clusters to the Submariner tunnel.
+
+``` go
+type SubmarinerGwRouteList struct {
+  metav1.TypeMeta `json:",inline"`
+  metav1.ListMeta `json:"metadata"`
+
+  Items []SubmarinerGwRoute `json:"items"`
+  }
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the next hop to reach the remote CIDRs
+    NextHop string `json:"nextHop"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+#### SubmarinerDefaultRoute
+
+This CR will be created when a remote cluster is connected. When more remote clusters are added the Remote CIDR list will be updated.
```

Comment:
Can we use endpoints itself instead of cluster CRDs?

Reply:
Done.